Assets/Features/Building/Scripts/Grid/Cursor.cs

```diff
@@ -0,0 +1,367 @@
+                {
+                    currentSelectedGroup.Add(new SubBuildItem(_selectedBuilding.BuildItems[0].Tile,
+                        new Vector3Int(x, min.y, (int)_selectedBuilding.BuildItems[0].GridPosition.Layer)));
+                    currentSelectedGroup.Add(new SubBuildItem(_selectedBuilding.BuildItems[0].Tile,
+                        new Vector3Int(x, max.y, (int)_selectedBuilding.BuildItems[0].GridPosition.Layer)));
+                }
+
+                for (var y = min.y; y < max.y + 1; y++)
+                {
+                    currentSelectedGroup.Add(new SubBuildItem(_selectedBuilding.BuildItems[0].Tile,
+                        new Vector3Int(min.x, y, (int)_selectedBuilding.BuildItems[0].GridPosition.Layer)));
+                    currentSelectedGroup.Add(new SubBuildItem(_selectedBuilding.BuildItems[0].Tile,
+                        new Vector3Int(max.x, y, (int)_selectedBuilding.BuildItems[0].GridPosition.Layer)));
+                }
+
+                Hover(Vector3Int.zero, currentSelectedGroup);
+            }
+            else
+            {
+                //hover in the default shape
+                Hover(position, _shape);
+            }
+        }
+
+        private void SingleBrush(Vector3Int position)
+        {
+            if (Input.GetMouseButtonDown(0))
+            {
+                //place the tiles in the shape of the selection
+                List<Vector3Int> positions = new List<Vector3Int>();
+                List<Tile> indices = new List<Tile>();
+                foreach (var selected in _selectedGroup)
+                {
+                    positions.Add(selected.GridPosition);
+                    indices.Add(selected.Tile);
+                }
+
+                _grid.SetGroup(positions, indices, _rotation);
+            }
+            else if (Input.GetMouseButtonDown(1))
+            {
+                //remove the tiles in the shape of the selection
+                _grid.Remove(position);
+            }
+            else
+            {
+                //hover and require all spaces to be free
+                Hover(position, _shape, true);
+            }
+        }
+
+        /// <summary>
+        /// Creates a tile on the positions the mouse is hold down
+        /// </summary>
+        /// <param name="position"></param>
+        private void DragBrush(Vector3Int position)
+        {
+            Hover(position, _shape);
+
+            if (Input.GetMouseButton(0))
+            {
+                _grid.Set(position, _selectedBuilding.BuildItems[0].Tile);
+            }
+            else if (Input.GetMouseButton(1))
+            {
+                //remove the tiles in the shape of the selection
+                _grid.Remove(position);
+            }
+        }
+
+        #endregion
+
+        private void Hover(Vector3Int position, List<SubBuildItem> shapes, bool requireAllAvailable = false, bool flipColors = false)
+        {
+            //Sets the offset of the whole grid
```

Comment:
Be consistent with the use of capitalized letters at the beginning of the sentence in your comments